app/services/kps/omu/adres.rb

```diff
@@ -15,27 +15,24 @@ def sorgula(queried_id_number)
             :adres_sorgula, message: message
           ).body[:adres_sorgula_response][:return][:sorgula_result][:sorgu_sonucu][:kimlik_noile_kisi_adres_bilgileri]
 
-          raise IdNumberError if response[:hata_bilgisi].present?
-
-          yerlesim_yeri = response[:yerlesim_yeri_adresi]
-          address_root = if yerlesim_yeri[:il_ilce_merkez_adresi].present?
-                           yerlesim_yeri[:il_ilce_merkez_adresi]
-                         elsif yerlesim_yeri[:koy_adresi].present?
-                           yerlesim_yeri[:koy_adresi]
-                         elsif yerlesim_yeri[:yurt_disi_adresi].present?
-                           yerlesim_yeri[:yurt_disi_adresi]
-                         end
-          # return a hash, ready to use for building an Address.
-          address_information = {
-            full_address: yerlesim_yeri[:acik_adres],
-            district_id: address_root[:ilce_kodu].to_i
-            # city: address_root[:il],
-            # city_id: address_root[:il_kodu],
-            # district: address_root[:ilce],
-            # neighbourhood: address_root[:mahalle],
-            # neighbourhood_id: address_root[:mahalle_kodu]
-          }
-          address_information
+          if response[:hata_bilgisi].present?
+            raise IdNumberError i
```

Comment:
Burada if/else bloğuna gerek yok. Kodun ilk hali doğru görünüyor. Bide `i` adında herhangi bir değişken yok sanırsam yazım hatası olmuş.

Bknz: https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/GuardClause

Reply:
Kodun ilk hali doğru değil. Bir yerde raise yapmak kodun geri kalanının çalışmasını önlemiyor. Örneğin:

```
x = nil
raise StandardError if x.nil?
x.capitalize
```

son satır olan `x.capitalize` yine çalışıp hata döndürecek.

GuardClause ise `if-end` blokları için geçerli. `if-else-end` için değil. zaten rubocop'ta hata vermiyor.

Reply:
Verdiğin örneği konsol üzerinde çalıştırdığımızda kodlar satır satır işlendiği ve bir fonksiyon içerisinde olmadığı için `x.capitalize` kodu da hata üretecektir. Fakat aşağıdaki örnekteki kodu denersen göreceksin ki sadece  `StandardError` üretecek ve `x.capitalize`  kodunu hiç çalıştırmayacak.

```ruby
def foo
  x = nil
  raise StandardError if x.nil?
  x.capitalize
end

foo # => StandardError: StandardError
```

Reply:
👍 ok. bunu merge ettim - sen fixlersin bi commitlerken